app/controllers/users_controller.rb

```diff
@@ -125,7 +125,7 @@ def extlogout
   def test_mail
     begin
       user = find_resource(:edit_users)
-      if params[:user_email].blank? && user.mail.blank?
+      if (params.has_key?(:user_email) && params[:user_email].blank?) || user.mail.blank?
```

Comment:
This check was actually incorrect before, since if the mail was changed to empty for a user that had a mail before this would return a success but try to send a mail with no recipient address.